File: Leetcode/String/681_Next_Closet_Time.cpp
class Solution {
public:
    string nextClosestTime(string time) {
        
    }

private:

	void dfs(int dep,const vector<int> )

	int toTime(int h, int m){
		return h*60+m;
	}

	int timeDiff(int t1, int t2){
		//Why?
		if(t1==t2) return INT_MAX

		return ((t2-t1)+24*60)%(24*60)
	}
};

File: Leetcode/String/681_Next_Closet_Time.py
class Solution(object):
    def nextClosestTime(self, time):
        """
        :type time: str
        :rtype: str
        """
        s=set(time)
        hour=int(time[0:2])
        minute=int(time[3:5])
        while True:
        	minute+=1
        	if minute==60:
        		minute=0
        		hour=0 if hour==23 else hour+1

        	time="%02d:%02d" %(hour,minute)
        	if set(time)<=s:
        		return time
        return -1
